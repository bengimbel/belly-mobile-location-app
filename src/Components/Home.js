import React, {Component} from 'react';
import {Platform, StyleSheet, Text, View} from 'react-native';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import BellyMapView from './MapView';
import fetchBasicData from '../Actions/FetchBasicData';
import fetchUserCoordinates from '../Actions/FetchUserCoordinates';
import _ from 'lodash';

type Props = {};

class Home extends Component<Props> {
    constructor(props){
        super(props)

        this.state = ({
            externalData: null,
            latitude: null,
            longitude: null,
            error: null
        })
    }

    componentDidMount() {
        navigator.geolocation.getCurrentPosition(
            (position) => {
              this.setState({
                latitude: position.coords.latitude,
                longitude: position.coords.longitude,
                error: null,
              }, () => {
                this.props.fetchBasicData(this.state.latitude, this.state.longitude)
              });
            },
            (error) => this.setState({ error: error.message }),
            { enableHighAccuracy: true, timeout: 20000, maximumAge: 1000 },
          )
      }

    check() {
        if(this.state.latitude){
            console.log(this.state.latitude, 'state.latitude')
        }
        if(this.props.userCoordinates.coordinates.coords){
            console.log(this.props.userCoordinates.coordinates.coords.latitude, 'latitude')
            console.log(this.props.userCoordinates.coordinates.coords.longitude, 'longitude')
        }
        if(this.props.basicData.data){
            console.log(this.props.basicData.data);
        }
    }

    render(){
        return (
            <View style={styles.container}>
                <BellyMapView />
                {
                    this.check()
                }
            </View>
        )
    }
}
function mapDispatchToProps(dispatch) {
    return bindActionCreators({ fetchBasicData, fetchUserCoordinates }, dispatch);
  }

const mapStateToProps = (state) => {
    return {
        basicData: state.basicData,
        userCoordinates: state.userCoordinates
    };
};


export default connect(mapStateToProps, mapDispatchToProps)(Home);

const styles = StyleSheet.create({
    container: {
      position: 'absolute',
      top: 0,
      left: 0,
      right: 0,
      bottom: 0,
      justifyContent: 'flex-end',
      alignItems: 'center'
    },
    map: {
      position: 'absolute',
      top: 0,
      left: 0,
      bottom: 0,
      right: 0
    }
  });