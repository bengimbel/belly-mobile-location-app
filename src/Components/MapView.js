import React, {Component} from 'react';
import MapView from 'react-native-maps';
import {Platform, StyleSheet, Text, View} from 'react-native';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
type Props = {};

class BellyMapView extends Component<Props> {
  constructor(props){
    super(props)

    this.state = ({
        latitude: null,
        longitude: null,
        data: null
    })
}

  renderMarkers() {
    console.log(this.props.data, 'DSDADADA')
    if(this.props.data){
      console.log(this.props.data, 'DATA I NEED')
      return (
        this.props.data.map((biz) => {
          return(
            <MapView.Marker
              key={biz.id}
              coordinate={{
                latitude: biz.coordinates.latitude,
                longitude: biz.coordinates.longitude,
                latitudeDelta: 0.1,
                longitudeDelta: 0.1
              }}
              title={biz.name}
              description={biz.name}
            />
          )
        })
      )

    } else{
      <Text>LOADING</Text>
    }
  }
    render() {
        return (
  
          <View style={styles.container}>
    
          {this.props.basicData.data.businesses &&
            
            <MapView style={styles.map}
              region={{
                latitude: this.props.lat,
                longitude: this.props.lon,
                latitudeDelta: 0.1,
                longitudeDelta: 0.1
              }}
              showsUserLocation
            >
              {
                this.renderMarkers()
              }
            </MapView>
          } 
          {!this.props.basicData.data.businesses &&
            <MapView style={styles.map}
              region={{
                latitude: 41.8781,
                longitude: -87.6298,
                latitudeDelta: 15.0,
                longitudeDelta: 15.0,
              }}
              showsUserLocation
            >
            </MapView>
          }
          </View>
        );
      }
}

const styles = StyleSheet.create({
    container: {
      position: 'absolute',
      top: 0,
      left: 0,
      right: 0,
      bottom: 0,
      justifyContent: 'flex-end',
      alignItems: 'center'
    },
    map: {
      position: 'absolute',
      top: 0,
      left: 0,
      bottom: 0,
      right: 0
    }
  });

  const mapStateToProps = (state) => {
    return {
        basicData: state.basicData,
        userCoordinates: state.userCoordinates
    };
};

export default connect(mapStateToProps, null)(BellyMapView);