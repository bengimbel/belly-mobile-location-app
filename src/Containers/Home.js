import React, {Component} from 'react';
import {Platform, StyleSheet, Text, View, Dimensions, TouchableOpacity, AsyncStorage} from 'react-native';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import BellyMapView from '../Components/MapView';
import BellyListView from '../Components/ListView';
import fetchBasicData from '../Actions/FetchBasicData';
import fetchUserCoordinates from '../Actions/FetchUserCoordinates';
import fetchCustomBusiness from '../Actions/FetchCustomBusiness';
import _ from 'lodash';
import { SearchBar } from 'react-native-elements';

class Home extends Component {
    constructor(props){
        super(props)

        this.state = ({
            externalData: null,
            myLatitude: null,
            myLongitude: null,
            error: null,
            termInput: '',
            activateMapButtonToggle: true
        })
        this.onTermInputChange = this.onTermInputChange.bind(this)
        this.onTermInputClear = this.onTermInputClear.bind(this)
        this.onSubmit = this.onSubmit.bind(this)
        this.renderMapButton = this.renderMapButton.bind(this)
    }

     async componentDidMount() {
        await AsyncStorage.getItem('storedData').then((value) => {
            let newValue = JSON.parse(value);
            console.log(newValue, 'parsedValue')
            this.setState({
                externalData: newValue
            })
         })
         await AsyncStorage.getItem('storedBasicDataLat').then((value) => {
            let newValue = JSON.parse(value);
            console.log(newValue, 'parsedValueLat')
            this.setState({
                myLatitude: newValue
            })
         })
         await AsyncStorage.getItem('storedBasicDataLon').then((value) => {
            let newValue = JSON.parse(value);
            console.log(newValue, 'parsedValueLon')
            this.setState({
                myLongitude: newValue
            })
         })

        navigator.geolocation.getCurrentPosition(
            (position) => {
                let lat = JSON.stringify(position.coords.latitude);
                let lon = JSON.stringify(position.coords.longitude);
                AsyncStorage.setItem('storedBasicDataLat', lat);
                AsyncStorage.setItem('storedBasicDataLon', lon);
                this.setState({
                myLatitude: position.coords.latitude,
                myLongitude: position.coords.longitude,
                error: null,
                }, () => {
                this.props.fetchBasicData(this.state.myLatitude, this.state.myLongitude).then(() => {
                    if(this.props.basicData.data.businesses){
                        let sortedBasicData = this.props.basicData.data.businesses.sort(function (a, b) {
                            return a.distance - b.distance;
                            });
                            console.log(sortedBasicData, 'sortedBasicData')
                            let stringifyValue = JSON.stringify(sortedBasicData)
                            AsyncStorage.setItem('storedData', stringifyValue);
                        this.setState({
                            externalData: sortedBasicData
                        })
                    }
                })
                });
            },
            (error) => this.setState({ error: error.message }),
            { enableHighAccuracy: true, timeout: 20000, maximumAge: 1000 },
            )
    }

    onTermInputChange(termInput){
        this.setState({
            termInput: termInput
        })
        console.log(this.state.termInput)
    }
    onTermInputClear(){
        this.setState({
            termInput: ''
        })
        console.log(this.state.termInput)
    }
    
    onSubmit() {
        this.props.fetchCustomBusiness(this.state.termInput, this.state.myLatitude, this.state.myLongitude).then(() => {
            if(this.props.customData.data.region){
                let sortedCustomData = this.props.customData.data.businesses.sort(function (a, b) {
                    return a.distance - b.distance;
                  });
                  AsyncStorage.setItem('storedData', JSON.stringify(sortedCustomData));
                this.setState({
                    externalData: sortedCustomData
                })
            }
        })
    }

    renderMapButton(){
        if(this.state.activateMapButtonToggle){
            return(
                 <TouchableOpacity
                    style={styles.toggleButton}
                    raised={true}
                    onPress={() => this.setState({
                        activateMapButtonToggle: !this.state.activateMapButtonToggle
                    })}
                >
                    <Text style={styles.toggleButtonText}>Show List</Text>
                </TouchableOpacity>
            )
        } else {
            return(
                <TouchableOpacity
                    style={styles.toggleButton}
                    raised={true}
                    onPress={() => this.setState({
                        activateMapButtonToggle: !this.state.activateMapButtonToggle
                    })}
                >
                    <Text style={styles.toggleButtonText}>Show Map</Text>
                </TouchableOpacity>
            )
        }

    }


    render(){
        return (
            <View style={styles.phoneContainer}>
                {this.state.activateMapButtonToggle ? 
                    <View style={styles.mainContainer}>
                        <View style={styles.searchContainer}>
                            <Text style={styles.title}>Locations</Text>
                            {
                                this.renderMapButton()
                            }
                            <SearchBar 
                                containerStyle={{backgroundColor: '#33A9E0', borderBottomColor: 'transparent', borderTopColor: 'transparent'}}
                                lightTheme
                                round
                                clearIcon={{ color: 'red' }}
                                inputContainerStyle={{backgroundColor: '#fff'}}
                                value={this.state.termInput}
                                onChangeText={this.onTermInputChange}
                                onClear={this.onTermInputClear}
                                onSubmitEditing={this.onSubmit}
                                style={{marginRight: 40, marginLeft: '40px', fontSize: 1}}
                                placeholder='Search by Keyword'
                                placeholderTextColor= '#33A9E0'
                                />
                        </View>
                        <BellyMapView
                            lat={this.state.myLatitude}
                            lon={this.state.myLongitude}
                            data={this.state.externalData}
                        /> 
                    </View>
                : 
                    <View style={styles.mainContainer}>
                        <View style={styles.searchContainer}>
                            <Text style={styles.title}>Locations</Text>
                            {
                                this.renderMapButton()
                            }
                        </View>
                        <BellyListView
                            data={this.state.externalData}
                        />
                    </View>
                }     
            </View>
        )
    }
}
function mapDispatchToProps(dispatch) {
    return bindActionCreators({ fetchBasicData, fetchUserCoordinates, fetchCustomBusiness }, dispatch);
  }

const mapStateToProps = (state) => {
    return {
        basicData: state.basicData,
        userCoordinates: state.userCoordinates,
        customData: state.customData
    };
};


export default connect(mapStateToProps, mapDispatchToProps)(Home);

const styles = StyleSheet.create({
    phoneContainer: {
        flex: 1,
        flexDirection: 'column',
        paddingTop: 25,
        backgroundColor: '#33A9E0'
    },
    mainContainer: {
        flex: 1,
        flexDirection: 'column',
        paddingTop: 25,
        backgroundColor: '#33A9E0'
    },
    searchContainer: {
        marginRight: 10,
        marginLeft: 10,
        backgroundColor: '#33A9E0'
    },
    toggleButton: {
        marginRight:10,
        marginLeft:10,
        marginTop:5,
        marginBottom:10,
        paddingTop:5,
        paddingBottom:5,
        backgroundColor:'#fff',
        borderRadius:10,
    },
    toggleButtonText: {
        color: '#33A9E0', 
        textAlign: 'center'
    },
    title: {
        fontSize: 26,
        color: '#fff',
        textAlign: 'center'
    }
  });