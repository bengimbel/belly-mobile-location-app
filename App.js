/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow
 */

import React, {Component} from 'react';
import { Provider } from 'react-redux';
import store from './src/Store';
import {Platform, StyleSheet, Text, View} from 'react-native';
import Home from './src/Components/Home';

const instructions = Platform.select({
  ios: 'Press Cmd+R to reload,\n' + 'Cmd+D or shake for dev menu',
  android:
    'Double tap R on your keyboard to reload,\n' +
    'Shake or press menu button for dev menu',
});

type Props = {};
export default class App extends Component<Props> {
  render() {
    return (
      <Provider store={store}>
        
          <Home />
        
      </Provider>

    );
  }
}

const styles = StyleSheet.create({
  container: {
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
    justifyContent: 'flex-end',
    alignItems: 'center'
  },
  // map: {
  //   position: 'absolute',
  //   top: 0,
  //   left: 0,
  //   bottom: 0,
  //   right: 0
  // }
});
