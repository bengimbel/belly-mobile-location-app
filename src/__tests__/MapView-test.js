import React from 'react';
import Home from '../Containers/Home';
import renderer from 'react-test-renderer';
import App from '../../App';
import MapView from '../Components/MapView';

test('renders correctly', () => {
  const tree = renderer.create(<App><Home><MapView /></Home></App>).toJSON();
  expect(tree).toMatchSnapshot();
});
