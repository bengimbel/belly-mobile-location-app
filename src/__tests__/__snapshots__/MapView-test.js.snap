// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#33A9E0",
      "flex": 1,
      "flexDirection": "column",
      "paddingTop": 25,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#33A9E0",
        "flex": 1,
        "flexDirection": "column",
        "paddingTop": 25,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#33A9E0",
          "marginLeft": 10,
          "marginRight": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontSize": 26,
            "textAlign": "center",
          }
        }
      >
        Locations
      </Text>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 10,
            "marginBottom": 10,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 5,
            "opacity": 1,
            "paddingBottom": 5,
            "paddingTop": 5,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#33A9E0",
              "textAlign": "center",
            }
          }
        >
          Show List
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#33A9E0",
            "borderBottomColor": "transparent",
            "borderBottomWidth": 1,
            "borderTopColor": "transparent",
            "borderTopWidth": 1,
            "padding": 8,
          }
        }
      >
        <View
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderBottomWidth": 0,
                "borderColor": "#86939e",
                "borderRadius": 15,
                "flexDirection": "row",
                "height": 30,
                "overflow": "hidden",
                "transform": Array [
                  Object {
                    "translateX": 0,
                  },
                ],
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "height": 40,
                  "justifyContent": "center",
                  "marginLeft": 8,
                }
              }
            >
              <View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    }
                  }
                  theme={
                    Object {
                      "colors": Object {
                        "divider": "#bcbbc1",
                        "error": "#ff190c",
                        "grey0": "#393e42",
                        "grey1": "#43484d",
                        "grey2": "#5e6977",
                        "grey3": "#86939e",
                        "grey4": "#bdc6cf",
                        "grey5": "#e1e8ee",
                        "greyOutline": "#bbb",
                        "primary": "#2089dc",
                        "searchBg": "#303337",
                        "secondary": "#8F0CE8",
                      },
                    }
                  }
                  underlayColor="white"
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": "#86939e",
                          "fontSize": 18,
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                        Object {
                          "fontFamily": "Material Design Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                    testID="iconIcon"
                  >
                    Ôçâ
                  </Text>
                </View>
              </View>
            </View>
            <TextInput
              allowFontScaling={true}
              onBlur={[Function]}
              onChangeText={[Function]}
              onClear={[Function]}
              onFocus={[Function]}
              onSubmitEditing={[Function]}
              placeholder="Search by Keyword"
              placeholderTextColor="#33A9E0"
              platform="default"
              style={
                Object {
                  "alignSelf": "center",
                  "color": "#86939e",
                  "flex": 1,
                  "fontSize": 18,
                  "height": 40,
                  "marginLeft": 10,
                }
              }
              testID="searchInput"
              underlineColorAndroid="transparent"
              value=""
            />
            <View
              style={
                Object {
                  "alignItems": "center",
                  "height": 40,
                  "justifyContent": "center",
                  "marginLeft": 15,
                  "marginRight": 8,
                }
              }
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              />
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "display": "flex",
          "flex": 1,
          "flexDirection": "row",
        }
      }
    />
  </View>
</View>
`;
